src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -0,0 +1,580 @@
+/**
+* Copyright (c) 2018 Zilliqa 
+* This source code is being disclosed to you solely for the purpose of your participation in 
+* testing Zilliqa. You may view, compile and run the code for that purpose and pursuant to 
+* the protocols and algorithms that are programmed into, and intended by, the code. You may 
+* not do anything else with the code without express permission from Zilliqa Research Pte. Ltd., 
+* including modifying or publishing the code (or any part of it), and developing or forming 
+* another public or private blockchain network. This source code is provided ‘as is’ and no 
+* warranties are given as to title or non-infringement, merchantability or fitness for purpose 
+* and, to the extent permitted by law, all liability for your use of the code is disclaimed. 
+* Some programs in this code are governed by the GNU General Public License v3.0 (available at 
+* https://www.gnu.org/licenses/gpl-3.0.en.html) (‘GPLv3’). The programs that are governed by 
+* GPLv3.0 are those programs that are located in the folders src/depends and tests/depends 
+* and which include a reference to GPLv3 in their program files.
+**/
+
+#include <boost/filesystem.hpp>
+
+#include "libUtils/DataConversion.h"
+#include "libUtils/JsonUtils.h"
+#include "libUtils/SysCommand.h"
+
+template<class MAP>
+AccountStoreSC<MAP>::AccountStoreSC()
+{
+	m_accountStoreAtomic = make_unique<AccountStoreAtomic<MAP>>(*this);
+}
+
+template<class MAP> void AccountStoreSC<MAP>::Init()
+{
+    AccountStoreBase<MAP>::Init();
+    m_curContractAddr.clear();
+}
+
+template<class MAP>
+bool AccountStoreSC<MAP>::UpdateAccounts(const uint64_t& blockNum,
+                                          const Transaction& transaction)
+{
+	LOG_MARKER();
+
+	lock_guard<mutex> g(m_mutexUpdateAccounts);
+
+    const PubKey& senderPubKey = transaction.GetSenderPubKey();
+    const Address fromAddr = Account::GetAddressFromPublicKey(senderPubKey);
+    Address toAddr = transaction.GetToAddr();
+    const uint256_t& amount = transaction.GetAmount();
+    uint256_t gasDeposit = transaction.GetGasLimit() * transaction.GetGasPrice();
```

Comment:
boost::multiprecision::uint256_t m_gasPrice;
    boost::multiprecision::uint256_t m_gasLimit;

We will have a security issue here. 
1. Integer overflow 
If (m_gasPrice * m_gasLimit) is larger than uint256, we will end up overflow and having smaller deposit
2. If gasPrice or gasLimit is unchecked and one of them is 0, we will end up with 0 deposit